var/spack/repos/builtin/packages/py-tensorflow/package.py

```diff
@@ -138,38 +144,51 @@ class PyTensorflow(Package, CudaPackage):
     depends_on('py-future', type='build', when='^python@:2')
 
     # Listed under REQUIRED_PACKAGES in tensorflow/tools/pip_package/setup.py
-    depends_on('py-absl-py@0.10:0', type=('build', 'run'), when='@2.4.0:')
+    depends_on('py-absl-py@0.4:', type=('build', 'run'), when='@2.7:')
+    depends_on('py-absl-py@0.10:0', type=('build', 'run'), when='@2.4.0:2.6')
     depends_on('py-absl-py@0.7.0:', type=('build', 'run'), when='@1.12.1,1.14:2.3')
     depends_on('py-absl-py@0.1.6:', type=('build', 'run'), when='@1.5:1.11')
-    depends_on('py-astunparse@1.6.3:1.6', type=('build', 'run'), when='@2.4.0:')
+    depends_on('py-astunparse@1.6.0:', type=('build', 'run'), when='@2.7:')
+    depends_on('py-astunparse@1.6.3:1.6', type=('build', 'run'), when='@2.4.0:2.6')
     depends_on('py-astunparse@1.6.3', type=('build', 'run'), when='@2.2:2.3')
     depends_on('py-astor@0.6.0:', type=('build', 'run'), when='@1.6:2.1')
     depends_on('py-backports-weakref@1.0:', type=('build', 'run'), when='@1.3: ^python@:3.3')
     depends_on('py-backports-weakref@1.0rc1', type=('build', 'run'), when='@1.2.0:1.2.1')
+    depends_on('py-clang@9.0.1:', type=('build', 'run'), when='@2.7:')
+    depends_on('py-clang@5.0.0:5.0', type=('build', 'run'), when='@2.6')
     depends_on('py-enum34@1.1.6:', type=('build', 'run'), when='@1.5: ^python@:3.3')
     depends_on('py-enum34@1.1.6:', type=('build', 'run'), when='@1.4.0:1.4.1')
-    depends_on('py-gast@0.3.3', type=('build', 'run'), when='@2.2:')
+    depends_on('py-gast@0.2.1:0.5.0', type=('build', 'run'), when='@2.7:')
+    depends_on('py-gast@0.4.0', type=('build', 'run'), when='@2.5:2.6')
+    depends_on('py-gast@0.3.3', type=('build', 'run'), when='@2.2:2.4')
     depends_on('py-gast@0.2.2', type=('build', 'run'), when='@1.15:2.1')
     depends_on('py-gast@0.2.0:', type=('build', 'run'), when='@1.6:1.14')
-    depends_on('py-google-pasta@0.2:0', type=('build', 'run'), when='@2.4.0:')
+    depends_on('py-google-pasta@0.1.1:', type=('build', 'run'), when='@2.7:')
+    depends_on('py-google-pasta@0.2:0', type=('build', 'run'), when='@2.4.0:2.6')
     depends_on('py-google-pasta@0.1.8:', type=('build', 'run'), when='@2.1:2.3')
     depends_on('py-google-pasta@0.1.6:', type=('build', 'run'), when='@1.14:2.0')
     depends_on('py-google-pasta@0.1.2:', type=('build', 'run'), when='@1.12.1')
     # propagate the mpi variant setting for h5py/hdf5 to avoid unexpected crashes
-    depends_on('py-h5py@2.10.0:2.10+mpi', type=('build', 'run'), when='@2.2:+mpi')
-    depends_on('py-h5py@2.10.0:2.10~mpi', type=('build', 'run'), when='@2.2:~mpi')
+    depends_on('py-h5py@2.9.0:3.1', type=('build', 'run'), when='@2.7:')
```

Comment:
The comment in the `setup.py` file is:
```
'h5py >= 2.9.0', # capped since 3.3.0 lacks py3.6
```
I interpreted that to mean that there was a cap around version 3.3, even though it is not explicit as it is for other capped specs. I can not recall now why I picked the cap to be 3.1 instead of 3.2 but I think the intent is for it to be capped. I certainly could be misinterpreting that though.

Reply:
I'm wondering if the cap used to be there, but was removed for some reason and the comment forgotten. Would have to dig through git blame.

Reply:
In general I try to keep the Spack recipe as close to possible as the `setup.py` recipe and only change things when I find that they don't build properly.

Reply:
I do as well. I know why I selected 3.1 as the cap. It was because that is what it was for tensorflow-2.6. That was the most recent value that I could find so I kept it. 

For tensorflow-2.6:
```
'h5py ~= 3.1.0'
```

Perhaps the comment is referring to why it was capped for earlier versions, although that would be odd from a context perspective.

Reply:
I can try a build with a newer h5py and see how it goes.